### DISCUSSION TITLE
Sitemap is empty

### DISCUSSION THREAD

**Discussion by adnanusstokyo** [spatie/laravel-sitemap#372] [Q&A]
After running the code:
`SitemapGenerator::create(url('/'))->writeToFile("sitemap.xml");`

A file named sitemap.xml is created on root of project, but it has only this content:
`<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
    <url>
        <loc>http://localhost/my_project_name</loc>
        <lastmod>2021-03-16T10:41:22+00:00</lastmod>
        <changefreq>daily</changefreq>
        <priority>0.8</priority>
    </url>
</urlset>`

**Comment by ousid** [comment]
Hey, @adnanusstokyo 

Can you show your implementation?

**Comment by guillaumedc** [comment]
I get an empty sitemap too.

I'm trying to generate the sitemap using this code:
```PHP
        SitemapGenerator::create(config('app.url'))
            ->configureCrawler(function (Crawler $crawler) {
                $crawler->ignoreRobots();
            })
            ->writeToFile(public_path('sitemap.xml'));
 ```
 
 The `app.url` config attribute points on the correct address.
 
 Do you know what can cause this?
 
 I'm using Sail to dev, could it be related?
 
 Thanks!

**Reply by fdjkgh580** [reply]
Thank you, it help me to fix my empty sitemap.xml

**Comment by SoloOrchid** [comment]
same issue on laravel breeze with vite and reactjs

code that creates empty sitemap

```php
SitemapGenerator::create(config('app.url'))->getSitemap()->writeToDisk('public', 'sitemap.xml');
```

**Comment by kul2002il** [comment]
Same issue with running php command in docker compose. It's happen because package request site by provided url. Make sure that site can requested from your script:
```php
use Illuminate\Support\Facades\Http;

dd(Http::get(config('app.url'))->status());
```

**Reply by alexander-feil** [reply]
Thank you, that helped me to track down this issue.
